gptel-context.el

```diff
@@ -119,16 +126,25 @@ context chunk.  This is accessible as, for example:
            (remove-p (< (prefix-numeric-value arg) 0))
 	   (action-fn (if remove-p
 			  #'gptel-context-remove
-			#'gptel-context-add-file)))
+			#'gptel-context-add-file))
+	   ;; Get git info just once for all files
+	   (git-cache (and (not remove-p)
+			   (gptel-context--build-git-cache default-directory))))
       (when (or remove-p (null dirs) (null confirm)
 		(y-or-n-p (format "Recursively add files from %d director%s? "
 				  (length dirs)
 				  (if (= (length dirs) 1) "y" "ies"))))
-	(mapc action-fn files))))
+	(if git-cache
+	    (dolist (file files)
+	      (funcall (if remove-p #'gptel-context-remove
+			 (lambda (f) (gptel-context-add-file f git-cache)))
+		       file))
+	  (mapc action-fn files)))))
    ;; If in an image buffer
    ((and (derived-mode-p 'image-mode)
-         (gptel--model-capable-p 'media)
-         (buffer-file-name))
+	 (gptel--model-capable-p 'media)
+	 (not (gptel-context--skip-file-p (buffer-file-name)))
+	 (buffer-file-name))
```

Comment:
Check for `(buffer-file-name)` first.  It can be `nil` and `(gptel-context--skip-file-p (buffer-file-name))` will throw an error.